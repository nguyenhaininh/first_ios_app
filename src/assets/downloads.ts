import { BookData } from './books';

export interface DownloadData {
  book: BookData;
  deliveryDate: string;
  endDate: string;
  size: number;
  downloaded: number;
}

const downloads: DownloadData[] = [
  {
    book: {
      title: 'mina 7月号',
      date: '2018/5/15',
      image: require('./images/dummy/cover-1.png')
    },
    deliveryDate: '2018/5/20',
    endDate: '2018/8/19',
    size: 84.74,
    downloaded: 100,
  }, {
    book: {
      title: 'non-no 5月号',
      date: '2018.5.15',
      image: require('./images/dummy/cover-3.png')
    },
    deliveryDate: '2018/5/20',
    endDate: '2018/8/19',
    size: 20.21,
    downloaded: 30,
  }, {
    book: {
      title: 'GISELe 6月号',
      date: '2018.5.15',
      image: require('./images/dummy/cover-33.png')
    },
    deliveryDate: '2018/5/20',
    endDate: '2018/8/19',
    size: 84.74,
    downloaded: 100,
  }, {
    book: {
      title: '&Premium 7月号',
      date: '2018.5.15',
      image: require('./images/dummy/cover-15.png')
    },
    deliveryDate: '2018/5/20',
    endDate: '2018/8/19',
    size: 20.21,
    downloaded: 30,
  }, {
    book: {
      title: 'Hanako 6月12日号',
      date: '2018.5.15',
      image: require('./images/dummy/cover-13.png')
    },
    deliveryDate: '2018/5/20',
    endDate: '2018/8/19',
    size: 84.74,
    downloaded: 100,
  }, {
    book: {
      title: 'FRaU 7月号',
      date: '2018.5.15',
      image: require('./images/dummy/cover-20.png')
    },
    deliveryDate: '2018/5/20',
    endDate: '2018/8/19',
    size: 20.21,
    downloaded: 30,
  },
];

export default downloads;
