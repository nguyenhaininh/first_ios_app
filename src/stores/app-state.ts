/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
export interface IAppState {
  informations?: IAppInformation[];
  isShowingMenu: boolean;
  mode: AppMode;
  tab: AppTab;
}

export enum AppMode {
  Init,
  Login,
  Member,
}

export enum AppTab {
  Feed,
  Ranking,
  Favorite,
  Download,
  Search
}

export interface IAppInformation {
  infoId: string;
  subject: string;
  description: string;
  date: string;
  important?: boolean;
}

export function createInitialAppState(): IAppState {
  return {
    isShowingMenu: false,
    mode: AppMode.Init,
    tab: AppTab.Feed,
  };
}
