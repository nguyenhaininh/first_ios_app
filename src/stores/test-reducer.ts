/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
import { createTypeReducer, isError } from '../common/utility/redux';
import { incrementAction, decrementAction, resetAction } from '../actions/test-actions';

export interface ITestState {
  counter: number;
  message?: string;
}

export function createInitialTestState(): ITestState {
  return { counter: 0 };
}

export const incrementReducer = incrementAction.reducer<ITestState>((state, action) => {
  if (isError(action)) {
    return {
      message: action.payload && action.payload.message,
    };
  }
  return {
    message: undefined,
    counter: state.counter += action.payload,
  };
});

export const decrementReducer = decrementAction.reducer<ITestState>((state, action) => ({
  counter: state.counter -= action.payload,
}));

export const resetReducer = resetAction.reducer<ITestState>((state, action) => {
  if (isError(action)) {
    return {
      message: action.payload && action.payload.message,
      counter: 0
    };
  }
  return {
    message: undefined,
    counter: 0
  };
});

export const testReducer = createTypeReducer(
  createInitialTestState,
  incrementReducer,
  decrementReducer,
  resetReducer,
);
