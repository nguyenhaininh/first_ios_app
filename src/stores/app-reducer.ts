/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
import { createTypeReducer, isError } from '../common/utility/redux';
import {
  initAppAction, fetchInfomartionsAction, setShowAppMenu, setTabAction, setModeAction,
} from '../actions/app-actions';
import { IAppState, createInitialAppState, AppMode } from './app-state';

export const appReducer = createTypeReducer<IAppState>(
  createInitialAppState,
  initAppAction.reducer((state, action) => {
    if (isError(action)) {
      throw action.error;
    }
    return {
      ...state,
      mode: AppMode.Login,
    };
  }),
  fetchInfomartionsAction.reducer((state, action) => isError(action) ? ({}) : ({
    informations: action.payload
  })),
  setShowAppMenu.reducer((state, action) => ({
    isShowingMenu: action.payload
  })),
  setTabAction.reducer((state, action) => ({
    tab: action.payload
  })),
  setModeAction.reducer((state, action) => ({
    mode: action.payload
  })),
);
