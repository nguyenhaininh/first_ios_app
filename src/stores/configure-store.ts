/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
import { createStore, combineReducers, applyMiddleware } from 'redux';

import { typePendingReducerSet, TypeReduxPendingState, createTypeReduxInitialState, typeReduxMiddleware } from '../common/utility/redux';

import { ITestState, testReducer, createInitialTestState } from './test-reducer';
import { appReducer } from './app-reducer';
import { createInitialAppState, IAppState } from './app-state';

export interface IStore extends TypeReduxPendingState {
  testState: ITestState;
  appState: IAppState;
}

export const rootReducer = combineReducers<IStore>({
  ...typePendingReducerSet,
  testState: testReducer,
  appState: appReducer,
});

const middlewares = [typeReduxMiddleware];

export const InitialState: IStore = {
  ...createTypeReduxInitialState(),
  testState: createInitialTestState(),
  appState: createInitialAppState(),
};

export const configureStore = (initialState: IStore = InitialState) => createStore(rootReducer, initialState, applyMiddleware(...middlewares));
