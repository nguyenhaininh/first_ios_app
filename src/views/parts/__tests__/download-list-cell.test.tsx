/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
import * as React from 'react';
import { Provider } from 'react-redux';
import DownloadCell from '../download-list-cell';

import * as renderer from 'react-test-renderer';
import { configureStore } from '../../../stores/configure-store';

import downloads from '../../../assets/downloads';

const onPress = () => {};

it('renders correctly', () => {
  const store = configureStore();
  const tree = renderer.create(
    <Provider store={store}>
      <DownloadCell data={downloads[0]} onPress={onPress}/>
    </Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
