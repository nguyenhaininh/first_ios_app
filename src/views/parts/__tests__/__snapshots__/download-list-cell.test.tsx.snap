// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": 156,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "padding": 15,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/assets/images/dummy/cover-1.png",
          }
        }
        style={
          Object {
            "backgroundColor": "#f7f7f7",
            "height": 126,
            "width": 100,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "marginLeft": 20,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000000",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginTop": 10,
            }
          }
        >
          mina 7月号
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000000",
              "fontSize": 11,
              "fontWeight": "bold",
              "marginTop": 10,
            }
          }
        >
          missing "en.book_issue_date_format" translation
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#999999",
              "fontSize": 11,
              "marginTop": 8,
            }
          }
        >
          [missing "en.delivery_date" translation](2018/5/20)[missing "en.end_date" translation](2018/8/19)
        </Text>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff20048",
                "flexDirection": "row",
                "height": 28,
                "paddingHorizontal": 10,
              },
              null,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#4f5b77",
                "fontSize": 11,
              }
            }
          >
            100% [missing "en.download_completed" translation]
          </Text>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#4f5b77",
                "fontSize": 11,
              }
            }
          >
            84.74MB
          </Text>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "right": 3,
          "top": 3,
          "width": 40,
        }
      }
    >
      <View>
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/images/delete_ico.png",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
